documents/service.go

```diff
@@ -49,3 +53,107 @@ type Service interface {
 	// ReceiveAnchoredDocument receives a new anchored document over the p2p layer, validates and updates the document in DB
 	ReceiveAnchoredDocument(model Model, headers *p2ppb.CentrifugeHeader) error
 }
+
+// service implements Service
+type service struct {
+	config Config
+	repo   Repository
+}
+
+// DefaultService returns the default implementation of the service
+func DefaultService(config config.Configuration, repo Repository) Service {
+	return service{repo: repo, config: config}
+}
+
+func getIds(model Model) ([]byte, []byte, error) {
+	cd, err := model.PackCoreDocument()
+
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return cd.DocumentIdentifier, cd.NextVersion, nil
+}
+
+func (s service) searchVersion(m Model) (Model, error) {
+
+	id, next, err := getIds(m)
+
+	nm, err := s.getVersion(id, next)
+
+	if err != nil {
+		// here the err is returned as nil because it is expected that the nextVersion
+		// is not available in the db at some stage of the iteration
+		return m, nil
+	}
+
+	if next != nil {
+		return s.searchVersion(nm)
+	}
+
+	return nm, nil
+}
+
+func (s service) GetCurrentVersion(documentID []byte) (Model, error) {
+	model, err := s.getVersion(documentID, documentID)
+	if err != nil {
+		return nil, errors.NewTypedError(ErrDocumentNotFound, err)
+	}
+	return s.searchVersion(model)
+
+}
+
+func (s service) GetVersion(documentID []byte, version []byte) (Model, error) {
+	return s.getVersion(documentID, version)
+}
+
+func (s service) CreateProofs(documentID []byte, fields []string) (*DocumentProof, error) {
+	return nil, nil
+}
+
+func (s service) CreateProofsForVersion(documentID, version []byte, fields []string) (*DocumentProof, error) {
+	return nil, nil
+}
+
+func (s service) DeriveFromCoreDocument(cd *coredocumentpb.CoreDocument) (Model, error) {
+	return nil, nil
+}
+
+func (s service) RequestDocumentSignature(contextHeader *header.ContextHeader, model Model) (*coredocumentpb.Signature, error) {
+	return nil, nil
+}
+
+func (s service) ReceiveAnchoredDocument(model Model, headers *p2ppb.CentrifugeHeader) error {
+	return nil
+}
+
+func (s service) Exists(documentID []byte) bool {
```

Comment:
We dont need this anymore, right ?

Reply:
Yes, right. We can remove it or leave it since we already have it. 
Maybe we need it in the future.
If we keep it, i will add some tests

Reply:
I am using it now for getting the currentVersion.
It's more elegant to use an exist instead of waiting for a db error to find the latest version